main.ts

```
@@ -1,132 +1,193 @@
+    constructor(expressionParse: ExpressionParserInterface) {
+        this.#parser = expressionParse;
     }
 
-    while (operators.length)
-        solveSimplifiedExpr(output, operators);
+    #solve() {
+        const secondNumber = Number(this.#output.pop());
+        const firstNumber = Number(this.#output.pop());
+        const operator = this.#operators.pop();
+
+        if (operator == null || !this.#operations.hasOwnProperty(operator))
```

Biome (1.9.4)

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)